src/deposit_monitor.rs

```diff
@@ -63,11 +84,49 @@ impl DepositMonitor {
 
         let bitcoin_client = self.context.bitcoin_client();
 
-        // TODO: cache results
-        let block_hash = bitcoin_client.get_block_hash(utxo.block_height)?;
-
-        // TODO: cache results
-        let tx_hex = bitcoin_client.get_raw_transaction_hex(&utxo.txid, &block_hash)?;
+        let cached_hash = self.hashes_cache.get(&utxo.block_height);
+        let block_hash = match cached_hash {
+            Some(hash) => {
+                tracing::debug!(
+                    "Using cached block hash for utxo at height {}: {}",
+                    utxo.block_height,
+                    hash
+                );
+                *hash
+            }
+            None => {
+                tracing::debug!(
+                    "Fetching block hash for utxo at height {}",
+                    utxo.block_height
+                );
+                let hash = bitcoin_client.get_block_hash(utxo.block_height)?;
+                self.hashes_cache.put(utxo.block_height, hash);
+                hash
+            }
+        };
```

Comment:
mmm thinking about this again, I don't think we can safely cache this. We should revert this (and change the comments from a TODO to a _we can't cache this_)

Reply:
hmm, yeas, make sense..

Reply:
wait sorry, I just meant this one (block height), we can still cache the other one (safely)